### PR TITLE
Audit Pocket Length/Volume/Weight Clothing

### PR DESCRIPTION
**Pull request by ANickelN**
**NEEDS TO BE MERGED BEFORE #48449**

#### Summary
Bugfixes "Add Pocket Lengths to Various Clothing Pockets"

#### Purpose of change

Makes longer items able to be put into longer pockets.

#### Describe the solution

Adds length to pockets, changed volume to match reality. 

#### Describe alternatives you've considered

Leaving clothing how it is, although this will prevent me from doing further work on item lengths.

#### Testing

Spawn the articles of clothing from the changes and ensure that their pockets have changed.

#### Additional context

Survivor/Nomad/Similar custom clothing was not touched in this pass, as it will take a lot more time and effort to go through and make sense of the pocket data.

### CODE REVIEW

**Comment by ANickelN** [comment]
Some things got pockets removed, like the Kimono and the Yukata.  I left a pseudo-pocket in to represent tucking some small objects into the sash.

File: data/json/items/armor/coats.json [hunk 1]
```diff
@@ -20,9 +20,27 @@
...
+        "pocket_type": "CONTAINER",
+        "max_contains_volume": "900 ml",
+        "max_contains_weight": "3 kg",
+        "max_item_length": "24 cm",
+        "moves": 80
+      },
+      {
+        "pocket_type": "CONTAINER",
+        "max_contains_volume": "3250 ml",
+        "max_contains_weight": "5 kg",
```
Comment (Maleclypse) [inline data/json/items/armor/coats.json:33]: Did you intend to change weight limit from 2kg to 5 kg?

File: data/json/items/armor/coats.json [hunk 2]
```diff
@@ -20,9 +20,27 @@
...
+        "max_item_length": "30 cm",
+        "moves": 80
+      },
+      {
+        "pocket_type": "CONTAINER",
+        "max_contains_volume": "3250 ml",
+        "max_contains_weight": "5 kg",
```
Comment (Maleclypse) [inline data/json/items/armor/coats.json:40]: Did you intend to change weight limit from 2kg to 5 kg?

File: data/json/items/armor/coats.json [hunk 3]
```diff
@@ -734,8 +1102,20 @@
       { "covers": [ "arm_l", "arm_r" ], "coverage": 90, "encumbrance": [ 15, 15 ] }
     ],
     "pocket_data": [
-      { "pocket_type": "CONTAINER", "max_contains_volume": "700 ml", "max_contains_weight": "3 kg", "moves": 80 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "700 ml", "max_contains_weight": "3 kg", "moves": 80 }
+      {
+        "pocket_type": "CONTAINER",
+        "max_contains_volume": "800 ml",
+        "max_contains_weight": "3 kg",
+        "max_item_length": "14 cm",
```
Comment (Maleclypse) [inline data/json/items/armor/coats.json:1109]: Out of curiosity why is this 800ml pocket 14 CM and the pocket on line 1073 above it is 800ml with 13CM max length?  Several places where this type of difference occurs.

File: data/json/items/armor/coats.json [hunk 4]
```diff
@@ -921,10 +1331,7 @@
       { "covers": [ "leg_l", "leg_r" ], "coverage": 90, "encumbrance": [ 2, 2 ] },
       { "covers": [ "arm_l", "arm_r" ], "coverage": 90, "encumbrance": [ 2, 2 ] }
     ],
-    "pocket_data": [
-      { "pocket_type": "CONTAINER", "max_contains_volume": "500 ml", "max_contains_weight": "3 kg", "moves": 80 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "500 ml", "max_contains_weight": "3 kg", "moves": 80 }
-    ],
+    "pocket_data": [ { "pocket_type": "CONTAINER", "max_contains_volume": "500 ml", "max_contains_weight": "1 kg", "moves": 80 } ],
```
Comment (Maleclypse) [inline data/json/items/armor/coats.json:1334]: Deleted a pocket here? No max length?

File: data/json/items/armor/coats.json [hunk 5]
```diff
@@ -948,10 +1355,7 @@
       { "covers": [ "leg_l", "leg_r" ], "coverage": 80, "encumbrance": [ 1, 1 ] },
       { "covers": [ "arm_l", "arm_r" ], "coverage": 80, "encumbrance": [ 1, 1 ] }
     ],
-    "pocket_data": [
-      { "pocket_type": "CONTAINER", "max_contains_volume": "500 ml", "max_contains_weight": "3 kg", "moves": 80 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "500 ml", "max_contains_weight": "3 kg", "moves": 80 }
-    ],
+    "pocket_data": [ { "pocket_type": "CONTAINER", "max_contains_volume": "500 ml", "max_contains_weight": "1 kg", "moves": 80 } ],
```
Comment (Maleclypse) [inline data/json/items/armor/coats.json:1358]: Deleted a pocket here? No max length?

File: data/json/items/armor/coats.json [hunk 6]
```diff
@@ -1639,10 +2271,27 @@
       { "covers": [ "arm_l", "arm_r" ], "coverage": 95, "encumbrance": [ 13, 13 ] }
     ],
     "pocket_data": [
-      { "pocket_type": "CONTAINER", "max_contains_volume": "350 ml", "max_contains_weight": "2 kg", "moves": 80 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "350 ml", "max_contains_weight": "2 kg", "moves": 80 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "250 ml", "max_contains_weight": "1 kg", "moves": 120 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "250 ml", "max_contains_weight": "1 kg", "moves": 120 }
```
Comment (Maleclypse) [inline data/json/items/armor/coats.json:1645]: Deleted a pocket here? 4 pockets originally.  3 pockets now. All pocket sizes changed so I'm assuming that you have a source.

File: data/json/items/armor/coats.json [hunk 7]
```diff
@@ -1667,10 +2316,27 @@
...
+        "max_item_length": "16 cm",
+        "moves": 80
+      },
+      {
+        "pocket_type": "CONTAINER",
+        "max_contains_volume": "600 ml",
+        "max_contains_weight": "1 kg",
+        "max_item_length": "13 cm",
+        "moves": 120
+      }
```
Comment (Maleclypse) [inline data/json/items/armor/coats.json:2339]: Deleted a pocket here? 4 pockets originally.  3 pockets now. All pocket sizes changed so I'm assuming that you have a source.

File: data/json/items/armor/coats.json [hunk 8]
```diff
@@ -1695,12 +2361,48 @@
...
+      {
+        "pocket_type": "CONTAINER",
+        "max_contains_volume": "1 L",
+        "max_contains_weight": "2 kg",
+        "max_item_length": "15 cm",
+        "moves": 80
+      },
+      {
+        "pocket_type": "CONTAINER",
+        "max_contains_volume": "900 ml",
```
Comment (Maleclypse) [inline data/json/items/armor/coats.json:2380]: Pocket sizes change in this set as well.

File: data/json/items/armor/legs_armor.json [hunk 9]
```diff
@@ -79,10 +79,34 @@
     "encumbrance": 24,
     "max_encumbrance": 30,
     "pocket_data": [
-      { "pocket_type": "CONTAINER", "max_contains_volume": "500 ml", "max_contains_weight": "1 kg", "moves": 80 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "500 ml", "max_contains_weight": "1 kg", "moves": 80 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "250 ml", "max_contains_weight": "1 kg", "moves": 100 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "250 ml", "max_contains_weight": "1 kg", "moves": 100 }
+      {
+        "pocket_type": "CONTAINER",
+        "max_contains_volume": "3250 ml",
```
Comment (Maleclypse) [inline data/json/items/armor/legs_armor.json:84]: Big changes here.

File: data/json/items/armor/suits_protection.json [hunk 10]
```diff
@@ -337,10 +397,10 @@
       { "covers": [ "arm_l", "arm_r" ], "coverage": 90, "encumbrance": [ 17, 17 ] }
     ],
     "pocket_data": [
-      { "pocket_type": "CONTAINER", "max_contains_volume": "700 ml", "max_contains_weight": "2 kg", "moves": 80 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "700 ml", "max_contains_weight": "2 kg", "moves": 80 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "250 ml", "max_contains_weight": "1 kg", "moves": 120 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "250 ml", "max_contains_weight": "1 kg", "moves": 120 }
+      { "pocket_type": "CONTAINER", "max_contains_volume": "800 ml", "max_contains_weight": "3 kg", "moves": 80 },
```
Comment (Maleclypse) [inline data/json/items/armor/suits_protection.json:400]: No length here?

**Review by Maleclypse — COMMENTED** [review]
In principle these changes look reasonable.  I've asked a number of questions and as I got further down I stopped asking why the changes from original sizes to new sizes, as I assume the answers for the early ones will also cover the later changes.  Thank you for doing this work.  These types of tasks are critical.

**Comment by Maleclypse** [comment]
> Some things got pockets removed, like the Kimono and the Yukata. I left a pseudo-pocket in to represent tucking some small objects into the sash.

This probably answers several of my questions.

File: data/json/items/armor/coats.json (hunk 1 above)
Comment (ANickelN) [reply data/json/items/armor/coats.json:33]: Yes, turnout coats should be able to hold both a large amount of items, and heavy items considering their purpose.

File: data/json/items/armor/coats.json (hunk 2 above)
Comment (ANickelN) [reply data/json/items/armor/coats.json:40]: Yes, this is the second of the "Patch Pockets" on some examples of a turnout coat I found.  They're usually around 10"x10", made of very sturdy material and should be easy for a firefighter to access with those thick gloves they wear.  The other smaller pocket is meant for a radio, and can't hold as much/as heavy.

File: data/json/items/armor/coats.json (hunk 3 above)
Comment (ANickelN) [reply data/json/items/armor/coats.json:1109]: That's the result of working on this for hours and hours and making copy-paste errors.  Thanks for pointing it out!

Comment (ANickelN) [reply data/json/items/armor/coats.json:1109]: Just to clarify, some jackets indeed do have just barely longer pockets than others, but items that are the same, but with a color difference should have the same stats.

File: data/json/items/armor/coats.json (hunk 4 above)
Comment (ANickelN) [reply data/json/items/armor/coats.json:1334]: Kimono was given a small pocket to represent being able to tuck some small items into the sash.  I've seen enough knife wielding kimono wearers in the movies to know they're hiding that thing somewhere.

File: data/json/items/armor/coats.json (hunk 5 above)
Comment (ANickelN) [reply data/json/items/armor/coats.json:1358]: Covered in the above ^^^

File: data/json/items/armor/coats.json (hunk 7 above)
Comment (ANickelN) [reply data/json/items/armor/coats.json:2339]: Waistcoats typically have two pockets near the abdomen, and a single breast pocket, never seen or heard of one with four pockets.

File: data/json/items/armor/coats.json (hunk 8 above)
Comment (ANickelN) [reply data/json/items/armor/coats.json:2380]: Matching the regular army jacket.  It should have the same pocket dimensions as the original jacket.

File: data/json/items/armor/legs_armor.json (hunk 9 above)
Comment (ANickelN) [reply data/json/items/armor/legs_armor.json:84]: The turnout pants I based this off of have the same 10x10 pockets as the coat does.  Two back pockets as well.

File: data/json/items/armor/suits_protection.json (hunk 10 above)
Comment (ANickelN) [reply data/json/items/armor/suits_protection.json]: All work and no play makes me a dull boy.

**Comment by ANickelN** [comment]
I would like to do another PR in the future regarding this, as these are not perfect measurements and calculations.  However, I feel a lot more comfortable with it as most pockets are within 15-20% of reality instead of being three or four times off what they should be, or clothes having pockets that they should not have, or not having pockets they should.

**Review by Maleclypse — APPROVED** [review]
Only one thing I missed in the first go round.   This looks good!

File: data/json/items/armor/legs_clothes.json [hunk 11]
```diff
@@ -17,12 +17,7 @@
     "coverage": 90,
     "encumbrance": 7,
     "max_encumbrance": 11,
-    "pocket_data": [
-      { "pocket_type": "CONTAINER", "max_contains_volume": "350 ml", "max_contains_weight": "1 kg", "moves": 80 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "350 ml", "max_contains_weight": "1 kg", "moves": 80 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "250 ml", "max_contains_weight": "1 kg", "moves": 100 },
-      { "pocket_type": "CONTAINER", "max_contains_volume": "250 ml", "max_contains_weight": "1 kg", "moves": 100 }
-    ],
+    "pocket_data": [ { "pocket_type": "CONTAINER", "max_contains_volume": "500 ml", "max_contains_weight": "1 kg", "moves": 400 } ],
```
[the remark this answers is not in the document]
Comment (ANickelN) [reply data/json/items/armor/legs_clothes.json]: That's also why it is 400 moves to retrieve the item, to represent loosening the cloth and grabbing it.

**Comment by ANickelN** [comment]
Ready for merging now that the mistake is fixed!

**Pull request merged**